extensions/mongodb/src/test/java/com/hazelcast/jet/mongodb/MongoDBSinkResilienceTest.java

```diff
@@ -0,0 +1,278 @@
+    public void testSink_networkTimeout() throws IOException {
+        sinkNetworkTest(true);
+    }
+
+    private void sinkNetworkTest(boolean timeout) throws IOException {
+        Config conf = new Config();
+        conf.addMapConfig(new MapConfig("*").setEventJournalConfig(new EventJournalConfig().setEnabled(true)));
+        conf.getJetConfig().setEnabled(true);
+        HazelcastInstance hz = createHazelcastInstance(conf);
+        JobRepository jobRepository = new JobRepository(hz);
+        createHazelcastInstance(conf);
+
+        final ToxiproxyClient toxiproxyClient = new ToxiproxyClient(toxi.getHost(), toxi.getControlPort());
+        final Proxy proxy = toxiproxyClient.createProxy("mongo", "0.0.0.0:8670", "mongo:27017");
+
+        final String databaseName = "networkCutoff";
+        final String collectionName = "testNetworkCutoff";
+        final String connectionViaToxi = "mongodb://" + toxi.getHost() + ":" + toxi.getMappedPort(8670);
+
+        IMap<Integer, Integer> sourceMap = hz.getMap(testName.getMethodName());
+
+        Pipeline pipeline = Pipeline.create();
+        pipeline.readFrom(Sources.mapJournal(sourceMap, START_FROM_OLDEST))
+                .withIngestionTimestamps()
+                .map(doc -> new Document("dummy", "test")
+                        .append("key", doc.getKey())
+                        .append("value", doc.getValue())
+                ).setLocalParallelism(2)
+                .writeTo(MongoDBSinks.builder("mongoSink", Document.class, () -> mongoClient(connectionViaToxi))
+                                     .into(databaseName, collectionName)
+                                     .preferredLocalParallelism(2)
+                                     .build());
+
+        Job job = invokeJob(hz, pipeline);
+        final AtomicBoolean continueCounting = new AtomicBoolean(true);
+        final AtomicInteger counter = new AtomicInteger(0);
+        spawn(() -> {
+            while (continueCounting.get()) {
+                int val = counter.incrementAndGet();
+                sourceMap.put(val, val);
+                sleep(random.nextInt(100));
+            }
+        });
+        waitForFirstSnapshot(jobRepository, job.getId(), 30, true);
+        sleep(1_000);
+
+        logger.info("Injecting toxis");
+        if (timeout) {
+            proxy.toxics().timeout("TIMEOUT", UPSTREAM, 0);
+            proxy.toxics().timeout("TIMEOUT_DOWNSTREAM", DOWNSTREAM, 0);
+        } else {
+            proxy.toxics().bandwidth("CUT_CONNECTION_DOWNSTREAM", DOWNSTREAM, 0);
+            proxy.toxics().bandwidth("CUT_CONNECTION_UPSTREAM", UPSTREAM, 0);
+        }
+        sleep(4_000 + random.nextInt(1000));
+        if (timeout) {
+            proxy.toxics().get("TIMEOUT").remove();
+            proxy.toxics().get("TIMEOUT_DOWNSTREAM").remove();
+        } else {
+            proxy.toxics().get("CUT_CONNECTION_DOWNSTREAM").remove();
+            proxy.toxics().get("CUT_CONNECTION_UPSTREAM").remove();
+        }
```

Comment:
@Patras3 could you point me to some tests which use the tool you researched? I would like to compare and use a single thing for this stuff.

Reply:
I guess it was this: https://github.com/hazelcast/hazelcast-enterprise/blob/e0308274cd9401a629eb3c4876f09f58d070b3d7/hazelcast-enterprise-it/enterprise-tests-using-docker/enterprise-tests-using-docker-utils/src/main/java/com/hazelcast/it/testsusingdocker/utils/SplitBrainUtil.java#L66

Quite similar in effect, except that tool is in EE repo

Reply:
Thanks @TomaszGaweda, yes it's the correct place.
Currently, there is no direct usage of ConnectionUtils in tests.
Used by ConnectionUtils `nftables` command is not available in the default mongo image. 
I just checked and yes, it can be installed by `apt-get update && apt-get install -y nftables` command but I don't know if that makes sense as it will need more adjustments because hz instance is not dockerized - to for example find our IP in terms of MongoDB container or to add a new command to block port (instead of IP).

`SplitBrainUtil`/`ConnectionUtils ` were designed for dockerized hazelcast instances where `nftables` is installed during building hazelcast docker image.

Reply:
Maybe we can move the resiliency test to EE? I also suspect @Patras3 will add more of these into the EE repo?